### PR TITLE
Add expectedOutputs to UnexpectedOutputException

### PR DESCRIPTION
**Pull request by jakemac53**
Fixes https://github.com/dart-lang/build/issues/552

### CODE REVIEW

File: build/lib/src/builder/exceptions.dart [hunk 1]
```diff
@@ -5,9 +5,11 @@ import '../asset/id.dart';
 
 class UnexpectedOutputException implements Exception {
   final AssetId assetId;
+  final Iterable<AssetId> expectedOutputs;
```
Comment (natebosch) [inline build/lib/src/builder/exceptions.dart]: [nit] can we call the `expected`? I think "Outputs" should be obvious in context.

File: build/lib/src/builder/exceptions.dart [hunk 2]
```diff
@@ -5,9 +5,11 @@ import '../asset/id.dart';
...
 
-  UnexpectedOutputException(this.assetId);
+  UnexpectedOutputException(this.assetId, {this.expectedOutputs});
 
   @override
-  String toString() => 'UnexpectedOutputException: $assetId';
+  String toString() => 'UnexpectedOutputException: $assetId'
+      '${expectedOutputs == null ? '' : '\nExpected one of: $expectedOutputs'}';
```
Comment (natebosch) [inline build/lib/src/builder/exceptions.dart]: [nit] "Expected only:" - "one of" might imply that we don't expect multiple, which we do allow.

File: build/lib/src/builder/exceptions.dart (hunk 1 above)
Comment (jakemac53) [reply build/lib/src/builder/exceptions.dart]: done

File: build/lib/src/builder/exceptions.dart (hunk 2 above)
Comment (jakemac53) [reply build/lib/src/builder/exceptions.dart]: done

**Comment by jakemac53** [comment]
Travis is having [issues today](https://www.traviscistatus.com/incidents/4vdl52d28hz3), going to run with a local `mono_repo presubmit` and submit assuming that passes.

**Pull request merged**